docs/how-to/how-to-run-docker-with-kata.md

```diff
@@ -0,0 +1,128 @@
+        name: dockersock
+      - mountPath: /var/lib/docker
+        name: docker
+  volumes:
+    - name: dockersock
+      emptyDir: {}
+    - name: docker
+      emptyDir:
+        medium: Memory
+```
+
+Inside the container you can view the mount:
+
+```console
+/ # mount | grep lib\/docker
+tmpfs on /var/lib/docker type tmpfs (rw,relatime)
+```
+
+As is mentioned in the comment encapsulating this code, using volatile memory for container storage backing is a risky and could be possibly wasteful on machines that do not have a lot of RAM.
+
+### Use a `loop` mounted disk
+
+Using a `loop`-mounted disk that is provisioned shortly before starting of the container workload is another approach that yields good performance. It was also [contributed](https://github.com/kata-containers/runtime/issues/1888#issuecomment-739057384) and the portion related to mounting the loopback disk is reproduced below:
+
+```yaml
+spec:
+  containers:
+    - name: docker
+      image: docker:dind
+      command: ["sh", "-c"]
+      args:
+      - if [[ $(df -PT /var/lib/docker | awk 'NR==2 {print $2}') == virtiofs ]]; then
+          apk add e2fsprogs &&
+          truncate -s 20G /tmp/disk.img &&
+          mkfs.ext4 /tmp/disk.img &&
+          mount /tmp/disk.img /var/lib/docker; fi &&
+        dockerd-entrypoint.sh;
+      securityContext:
+        privileged: true
+```
+
+Note that `loop` mounted disks are often sparse, which means they *do not* take up the full amount of space that has been provisioned. This solution seems to produce the best performance and flexibility, at the expense of increased complexity and additional required setup.
+
+### Build a custom kernel/Wait for upstream resolution
+
+`virtio-fs` may be supported as an upper layer for `virtio-fs` in future kernel releases, and it's possible to [modify the kernel](https://github.com/kata-containers/runtime/issues/1888#issuecomment-616872558) (in addition to applying the earlier mentioned mailing list patch) to support using `virtio-fs` as an upper. Note that if you modify your kernel and use `virtio-fs` you may require [additional changes](https://github.com/kata-containers/runtime/issues/1888#issuecomment-739057384) for decent performance and to address other issues.
+
+## References
+
+The solutions proposed in this document are an amalgamation of thoughtful contributions from the Kata Containers community. Find links to discussion and the fruits therein below:
+
+- [How to run Docker in Docker with Kata Containers (`runtime`#2474)](https://github.com/kata-containers/kata-containers/issues/2474)
+- [Does Kata-conatiner support AUFS/OverlayFS? (`runtime`#2493)](https://github.com/kata-containers/runtime/issues/2493)
+- [Unable to start docker in docker with virtio-fs (`runtime`#1888)](https://github.com/kata-containers/runtime/issues/1888)
+- [Not using native diff for overlay2 (`runtime`#1429)](https://github.com/kata-containers/runtime/issues/1429#issuecomment-477385283)
```

Comment:
It's unclear what `runtime` means in this references. I assumed it referred to the GitHub repository, but if so, the first one is incorrect. It might be clearer to either remove the part in brackets or maybe use tags like `(Kata 1.x)` and `(Kata 2.x)`?

Reply:
I can remove `runtime` and replace it with "Issue" -- what I meant to list here were links to related issues and discussion around this for context.

Reply:
I've replaced the occurrences and rewrote that section a little bit